jgrapht-core/src/main/java/org/jgrapht/generate/LinearizedChordDiagramGraphGenerator.java

```diff
@@ -0,0 +1,149 @@
+ * node is proportional to its degree.
+ * 
+ * @author Dimitrios Michail
+ * @since February 2017
+ * 
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public class LinearizedChordDiagramGraphGenerator<V, E>
+    implements GraphGenerator<V, E, V>
+{
+    private final Random rng;
+    private final int m;
+    private final int n;
+
+    /**
+     * Constructor
+     * 
+     * @param n number of nodes
+     * @param m number of edges of each new node added during the network growth
+     * @throws IllegalArgumentException in case of invalid parameters
+     */
+    public LinearizedChordDiagramGraphGenerator(int n, int m)
+    {
+        this(n, m, new Random());
+    }
+
+    /**
+     * Constructor
+     * 
+     * @param n number of nodes
+     * @param m number of edges of each new node added during the network growth
+     * @param seed seed for the random number generator
+     * @throws IllegalArgumentException in case of invalid parameters
+     */
+    public LinearizedChordDiagramGraphGenerator(int n, int m, long seed)
+    {
+        this(n, m, new Random(seed));
+    }
+
+    /**
+     * Constructor
+     * 
+     * @param n number of nodes
+     * @param m number of edges of each new node added during the network growth
+     * @param rng the random number generator to use
+     * @throws IllegalArgumentException in case of invalid parameters
+     */
+    public LinearizedChordDiagramGraphGenerator(int n, int m, Random rng)
+    {
+        if (n <= 0) {
+            throw new IllegalArgumentException("invalid number of nodes: must be positive");
+        }
+        this.n = n;
+        if (m <= 0) {
+            throw new IllegalArgumentException("invalid edges per node (" + m + " <= 0");
+        }
+        this.m = m;
+        this.rng = Objects.requireNonNull(rng, "Random number generator cannot be null");
+    }
+
+    /**
+     * Generates an instance.
+     * 
+     * @param target the target graph, which must allow self-loops and parallel edges
+     * @param vertexFactory the vertex factory
+     * @param resultMap not used by this generator, can be null
+     * @throws IllegalArgumentException if the graph does not allow self-loops or parallel edges
+     */
+    @Override
+    public void generateGraph(
+        Graph<V, E> target, VertexFactory<V> vertexFactory, Map<String, V> resultMap)
+    {
+        /*
+         * Add nodes by maintaining a list with vertex multiplicity equal to its degree for sampling
+         * purposes.
+         */
+        List<V> nodes = new ArrayList<>(2 * n * m);
+        for (int t = 0; t < n; t++) {
+            // add node
+            V vt = vertexFactory.createVertex();
+            if (!target.addVertex(vt)) {
+                throw new IllegalArgumentException("Invalid vertex factory");
+            }
+
+            // add edges
+            for (int j = 0; j < m; j++) {
+                // add outward half degree before sampling
```

Comment:
You need to read the paper. Still, you understood right. They split each edge into two conceptual parts in order to be able to sample self-loops.